Upstream: needs work
Description: Perhaps cleanup a bit and submit to upstream
 And we need to force *FLAGS_FOR_BUILD to something,
 otherwise, those flags will be empty and make will pick up the *FLAGS
 environment during build.
--- a/configure.ac
+++ b/configure.ac
@@ -5836,10 +5836,11 @@ if test "$cross_compiling" = "yes"; then
     test -d config_build && cp -p config_build/*.h CONF-FOR-BUILD/config_host 2>/dev/null
     (
     unset COM USING_X11 OS CPUNAME
-    unset CC CXX SYSBASE CFLAGS
+    unset CC CXX SYSBASE CFLAGS CXXFLAGS CPP CPPFLAGS LD LDFLAGS
     unset AR LD NM OBJDUMP PKG_CONFIG RANLIB READELF STRIP
     unset CPPUNIT_CFLAGS CPPUNIT_LIBS
     unset LIBXML_CFLAGS LIBXML_LIBS LIBXSLT_CFLAGS LIBXSLT_LIBS XSLTPROC
+    unset PYTHON_CFLAGS PYTHON_LIBS
     unset PKG_CONFIG_LIBDIR PKG_CONFIG_PATH
     if test -n "$CC_FOR_BUILD"; then
         export CC="$CC_FOR_BUILD"
@@ -5849,6 +5850,12 @@ if test "$cross_compiling" = "yes"; then
         export CXX="$CXX_FOR_BUILD"
         CXX_BASE=`first_arg_basename "$CXX"`
     fi
+    export CFLAGS="$CFLAGS_FOR_BUILD"
+    export CXXFLAGS="$CXXFLAGS_FOR_BUILD"
+    export CPP="$CPP_FOR_BUILD"
+    export CPPFLAGS="$CPPFLAGS_FOR_BUILD"
+    export LD="$LD_FOR_BUILD"
+    export LDFLAGS="$LDFLAGS_FOR_BUILD"
     test -n "$PKG_CONFIG_FOR_BUILD" && export PKG_CONFIG="$PKG_CONFIG_FOR_BUILD"
     cd CONF-FOR-BUILD
 
